src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3448,9 +3448,17 @@ public long GetIndexEtag(QueryOperationContext context, QueryMetadata q)
                 using (_contextPool.AllocateOperationContext(out TransactionOperationContext indexContext))
                 {
                     using (indexContext.OpenReadTransaction())
-                    using (context.OpenReadTransaction())
                     {
-                        return CalculateIndexEtag(context, indexContext, q, IsStale(context, indexContext));
+                        DocumentsTransaction  d = null;
+                        if (context.Documents.HasTransaction == false)
```

Comment:
This is incorrect. QueryOperationContext operates on both Server-Storage and Database-Storage transactions. IIRC there is a method there 'AreTransactionsOpened' to check that.